src/Tools/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/SyntaxVisualizerControl.xaml.cs

```diff
@@ -683,11 +712,23 @@ private static TreeViewItem FindTreeViewItem(DependencyObject source)
         #region Event Handlers
         private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEventArgs<object> e)
         {
-            if (treeView.SelectedItem != null)
+            // Restore the regular colors on the newly unselected item
+            var previousSelection = _currentSelection;
+            if (previousSelection != null)
+            {
+                previousSelection.Foreground = _currentSelectionUnselectedForeground;
+            }
+
+            // Remember the newly selected item's specified foreground color and then clear it to
+            // allow the ControlTemplate to correctly set contrasting colors.
+            _currentSelection = (TreeViewItem)treeView.SelectedItem;
+            if (_currentSelection != null)
             {
```

Comment:
@333fred There was a merge conflict here, can you check if this was resolved correctly?